context/providers/featureFlagsProvider.tsx

```diff
@@ -0,0 +1,30 @@
+import { FEATURE_FLAGS_QUERY } from "@/graphql/client"
+import { useQuery } from "@apollo/client"
+import React, { useMemo } from "react"
+import { FeatureFlagsContext } from "../featureFlagsContext"
+import { FeatureFlagsContextProps } from "../featureFlagsContext"
+
+export const FeatureFlagsContextProvider: React.FC<{
+  children: React.ReactNode
+}> = ({ children }) => {
+  const { loading, data } = useQuery(FEATURE_FLAGS_QUERY)
+
+  const contextValue = useMemo(() => {
+    const featureFlagsContextProps = {} as FeatureFlagsContextProps
+    data?.featureFlags.forEach(({ name, isEnabled, requiredRole }) => {
+      featureFlagsContextProps[name as keyof FeatureFlagsContextProps] = {
+        isEnabled,
+        requiredRole,
+      }
+    })
+    return featureFlagsContextProps
+  }, [data])
```

Comment:
this component runs the feature flags query and adds each flag to the global state